compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -471,6 +471,21 @@ generateSrcFromLf env = noLoc mod
                 (intR,tagR) = T.span isDigit (T.reverse name)
             in (T.reverse tagR, readMay (T.unpack (T.reverse intR)))
 
+        getOverlapMode :: LF.ExprValName -> Maybe (Located OverlapMode)
+        getOverlapMode (LF.ExprValName name) = do
+            dval <- NM.lookup (overlapModeName name) (LF.moduleValues (envMod env))
+            LF.EBuiltin (LF.BEText mode) <- Just (LF.dvalBody dval)
+            modeFn <- MS.lookup mode overlapModeFns
+            Just (noLoc (modeFn NoSourceText))
+
+        overlapModeFns :: MS.Map T.Text (SourceText -> OverlapMode)
+        overlapModeFns = MS.fromList
```

Comment:
Maybe worth defining the mapping here and the one in the LF conversion somewhere close to each other or at least add a comment that the two must stay in sync.

Reply:
A roundtrip test might also be a good idea and avoids the need to write four tests just to make sure we translate strings correctly :slightly_smiling_face:

Reply:
That's a good idea. (I think I'll do that as a separate refactoring PR since it involves touching more code.)